client/src/components/HOC/withInteractionApi.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import axios from 'axios';
+
+var withInteractionsApi = function (WrappedComponent, widget) {
+
+  return class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.sendInteraction = this.sendInteraction.bind(this);
+      this.formatBody = this.formatBody.bind(this);
+    }
+
+    formatBody(method, apiRoute, params = {}, data = {}) {
```

Comment:
As discussed we may have to change this, but Thanks so much for taking this task on. Much appreciated Ash!

Reply:
No problem. It's my pleasure.